packages/caravan-health/src/privacy.ts

```diff
@@ -0,0 +1,285 @@
+import { BlockchainClient } from "@caravan/clients";
+import type { UTXO, Transaction } from "@caravan/clients/src/client";
+
+export interface WalletUTXOs {
+  address: string;
+  utxos: Array<UTXO>;
+}
+
+/*
+The methodology for calculating a privacy score (p_score) for Bitcoin transactions based 
+on the number of inputs and outputs is the primary point to define wallet health for privacy. 
+The score is further influenced by several factors such as address reuse, 
+address types and UTXO set fingerprints etc.
+
+More on the algorithms for scoring privacy: <TODO : Add Link to Blog Post>
+*/
+
+// A normalizing quantity that increases the score by a certain factor in cases of self-payment.
+// More about deniability : https://www.truthcoin.info/blog/deniability/
+const DENIABILITY_FACTOR = 1.5;
+
+/*
+The p_score is calculated by evaluating the likelihood of self-payments, the involvement of 
+change outputs and the type of transaction based on number of inputs and outputs.
+
+We have 5 categories of transaction type
+- Sweep Spend
+- Simple Spend
+- UTXO Fragmentation
+- Consolidation
+- CoinJoin
+*/
+enum SpendType {
+  SweepSpend = "SweepSpend",
+  SimpleSpend = "SimpleSpend",
+  UTXOFragmentation = "UTXOFragmentation",
+  Consolidation = "Consolidation",
+  MixingOrCoinJoin = "MixingOrCoinJoin",
+}
+
+function SpendTypeScores(
+  spendType: SpendType,
+  numberOfInputs: number,
+  numberOfOutputs: number,
+): number {
+  switch (spendType) {
+    case SpendType.SweepSpend:
+      return 1 / 2;
+    case SpendType.SimpleSpend:
+      return 4 / 9;
+    case SpendType.UTXOFragmentation:
+      return 2 / 3 - 1 / numberOfOutputs;
+    case SpendType.Consolidation:
+      return 1 / numberOfInputs;
+    case SpendType.MixingOrCoinJoin:
+      let x = Math.pow(numberOfOutputs, 2) / numberOfInputs;
+      return (0.75 * x) / (1 + x);
```

Comment:
I would imagine that the linter would expect a default case for a switch statement.